src/traces/funnel/defaults.js

```diff
@@ -0,0 +1,90 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+
+var handleGroupingDefaults = require('../bar/defaults').handleGroupingDefaults;
+var handleText = require('../bar/defaults').handleText;
+var handleXYDefaults = require('../scatter/xy_defaults');
+var attributes = require('./attributes');
+var Color = require('../../components/color');
+
+function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    var len = handleXYDefaults(traceIn, traceOut, layout, coerce);
+    if(!len) {
+        traceOut.visible = false;
+        return;
+    }
+
+    coerce('orientation', (traceOut.x && !traceOut.y) ? 'h' : 'v');
```

Comment:
Remember, we should make the default `orientation: 'h'` - unless `(!traceOut.x && traceOut.y)`

Reply:
Done in https://github.com/plotly/plotly.js/pull/3817/commits/b93e3a5a83b6561ac6258a59f274b5fc87630c3e.
What about `autorange: "reversed"`?

Reply:
> Done in b93e3a5.

Nice! Can you mention something about the distinct behaviour in the funnel `orientation` attribute `description`?

Reply:
> What about `autorange: "reversed"`?

That item is still unresolved.

Reply:
Now resolved.